quotientai/tracing.ts

```diff
@@ -0,0 +1,338 @@
+    process.on('SIGINT', () => {
+      cleanupAllInstances().then(() => process.exit(0));
+    });
+
+    process.on('SIGTERM', () => {
+      cleanupAllInstances().then(() => process.exit(0));
+    });
+
+    process.on('beforeExit', () => {
+      cleanupAllInstances();
+    });
+
+    // Handle uncaught exceptions and unhandled rejections
+    process.on('uncaughtException', (error) => {
+      logError(new Error(`Uncaught exception: ${error}`));
+      cleanupAllInstances().then(() => process.exit(1));
+    });
+
+    process.on('unhandledRejection', (reason, promise) => {
+      logError(new Error(`Unhandled rejection at: ${promise}, reason: ${reason}`));
+      cleanupAllInstances().then(() => process.exit(1));
+    });
+  }
+
+  init(config: TracingConfig): void {
+    // Validate inputs
+    if (!config.app_name || typeof config.app_name !== 'string') {
+      logError(new Error('app_name must be a non-empty string'));
+      return;
+    }
+    if (!config.environment || typeof config.environment !== 'string') {
+      logError(new Error('environment must be a non-empty string'));
+      return;
+    }
+    if (config.instruments && !Array.isArray(config.instruments)) {
+      logError(new Error('instruments must be an array'));
+      return;
+    }
+
+    this.appName = config.app_name;
+    this.environment = config.environment;
+    this.instruments = config.instruments || [];
+    this.endpoint = config.endpoint || this.endpoint;
+    this.headers = config.headers || {};
+
+    // Initialize tracer with OTLP exporter
+    this.setupTracer();
+  }
+
+  private setupTracer(): void {
+    if (this.isConfigured || !this.appName || !this.environment) {
+      return;
+    }
+
+    try {
+      const defaultHeaders = {
+        Authorization: `Bearer ${this.client.apiKey}`,
+        // Note: Content-Type header is automatically set by the protobuf exporter
+        ...this.headers,
+      };
+
+      if (process.env.OTEL_EXPORTER_OTLP_HEADERS) {
+        try {
+          const envHeaders = JSON.parse(process.env.OTEL_EXPORTER_OTLP_HEADERS);
+          if (typeof envHeaders === 'object' && envHeaders !== null) {
+            Object.assign(defaultHeaders, envHeaders);
+          }
+        } catch (error) {
+          console.warn('Failed to parse OTEL_EXPORTER_OTLP_HEADERS, using default headers...');
+        }
+      }
+
+      const traceExporter = new OTLPTraceExporter({
+        url: this.endpoint,
+        headers: defaultHeaders,
+        keepAlive: true,
+      });
+
+      const resource = resourceFromAttributes({
+        [QuotientAttributes.APP_NAME]: this.appName,
+        [QuotientAttributes.ENVIRONMENT]: this.environment,
+      });
+
+      this.sdk = new NodeSDK({
+        resource,
+        spanProcessor: new BatchSpanProcessor(traceExporter),
+      });
+
+      // Start the SDK
+      this.sdk.start();
+
+      this.handleManualInstrumentation();
+
+      this.tracer = trace.getTracer('quotient-tracer', '0.1.0');
```

Comment:
We should call this `quotient.sdk.typescript` . See this as an example: https://github.com/quotient-ai/quotient-python/blob/40a40d4af62c8aaf2c333fbfddbf374dafc7eb0e/quotientai/_constants.py#L2